src/mem.c

```diff
@@ -104,6 +104,24 @@ void ofi_mem_fini(void)
 	free(page_sizes);
 }
 
+size_t ofi_get_mem_size(void)
+{
+	long page_cnt, page_size;
+	size_t mem_size;
+
+	page_cnt = sysconf(_SC_PHYS_PAGES);
+	page_size = ofi_get_page_size();
+
+	if (page_cnt <= 0 || page_size <= 0)
+		return 0;
+
+	mem_size = (size_t) page_cnt * (size_t) page_size;
```

Comment:
I could be wrong, but this wouldn't factor in hugepages on systems where that's configured and the bufpool uses it. It shouldn't drastically impact the intent of this patch since the cache would mostly deal with application-level registrations than internal ones..

Reply:
The intent here is to determine the amount of memory available in the system.  Both page_cnt and page_size are coming from sysconf(), so I assume that they are related to each other and can be used in this calculation.

Reply:
👍